### PR TITLE
Track

### PR DESCRIPTION
**Pull request by mikkelblyme**
**FIGMA link: https://www.figma.com/file/Gmk9SRFJvVBkYmNjLVwKOF/Track-univers---DDS.dk?node-id=1%3A3**

# What
The "TRACK" universe is a place for current content on spejder.dk will live in the future on dds.dk
This is a huge pr (sorry). I decided to make one big PR due to limited PR envs and the fact that almost each tasks was dependent on the prior task. I didn't rebase it as I was afraid of breaking the whole thing. Let me know if you would like that, then I'll give it a go.

# How
In order for the editor to determine what content should be in the track-universe or not, they use tagging. There exists a "Category" vocabulary where the client and I decided to put the "TRACK" tags. I've created a "Track" tag and added the four track categories under that parent tag:
The weight are like so:
- Track (this is disabled and the editors thus can't select it in the category tag dropdown on nodes)
    - Adventurespejd
    - Patruljeliv
    - Skills
    - Arrangementer
(the order doesn't matter)

By doing it this way, I can check if the chosen tag has a parent with the correct id. Like so: 
https://github.com/reload/ddsdk2016/blob/track/web/modules/custom/dds_track/dds_track.module#L253-L274

The function in above link, return a boolean whether the current node is in a track-context. I then use that boolean to set different stuff. 

Some "important" html classes: 
- I set a `page-type-universe` html class on the body that allows me to control elements at a high level
- I set a `universe-track-${THE TRACK TAG CHOSEN}` (example `universe-track-adventurespejd`) to help determine icons, color and so on.
- node-teasers also get classes so I can show the little spinning icon, add some styling etc.

---

#### Where should the reviewer start?
Pretty much all logic exists in `dds_track/dds_track.module` to take a glimpse in there :)

### CODE REVIEW

File: web/themes/custom/mungo/assets_src/sass/track/_section-page.scss [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+      top: -$spacing-large;
+      left: 50%;
+      bottom: initial;
+      transform: translateX(-50%);
+
+      @include media-medium {
+        left: 5%;
+        transform: translateX(0);
+      }
+      // @include header-with-bg(subtitle)
```
Comment (rasben) [inline web/themes/custom/mungo/assets_src/sass/track/_section-page.scss:96]: Looks like a mistake.

File: web/themes/custom/mungo/assets_src/sass/track/_section-page.scss [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+      }
+    }
+    @include header-with-bg($targetSelector: subtitle);
+
+    // Icons.
+    // Fill the t-logo "t" with the color of the current chosen track.
+    @each $track, $track-color in $track-colors {
+      // We don't want to set a bg for our T-icon if the track chosen is "arrangementer",
+      // as the bg color for "arrangementer" is already yellow.
+      @if ($track != "arrangementer") {
```
Comment (rasben) [inline web/themes/custom/mungo/assets_src/sass/track/_section-page.scss:213]: It is a bit ugly to use a danish word in the CSS, but i assume you have no choice in terms of using a TID instead?

File: web/themes/custom/mungo/mungo.theme [hunk 3]
```diff
@@ -85,7 +89,7 @@ function mungo_preprocess_field(&$variables) {
     }
   }
   elseif (in_array($field_name, ['field_2col_deck_col1', 'field_2col_deck_col2'])) {
-    /** @var Paragraph $paragraph */
+    /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
     $paragraph = $variables['element']['#object'];
```
Comment (rasben) [inline web/themes/custom/mungo/mungo.theme:93]: I really think using instance of + return would be more stable here.

File: web/themes/custom/mungo/mungo.theme [hunk 4]
```diff
@@ -317,6 +349,8 @@ function mungo_preprocess_paragraph(&$variables) {
       break;
 
     case 'situation_links':
+      /** @var \Drupal\node\NodeInterface $node */
+      $node = \Drupal::routeMatch()->getParameter('node');
```
Comment (rasben) [inline web/themes/custom/mungo/mungo.theme:353]: See my comments earlier about `instance of`

File: web/themes/custom/mungo/mungo.theme [hunk 5]
```diff
@@ -1081,6 +1120,9 @@ function _mungo_2col_size_option_to_text(int $value) {
 function _mungo_preprocess_deck_paragraph($type, &$variables) {
   $variables['cols_attributes'] = new Attribute();
 
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = \Drupal::routeMatch()->getParameter('node');
```
Comment (rasben) [inline web/themes/custom/mungo/mungo.theme:1124]: See my comments earlier about `instance of`

File: web/themes/custom/mungo/templates/paragraphs/paragraph--embed.html.twig [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  </div>
+  <div class="embed-code--wrapper">
+    <div
+      class="embed-code {% if cookie_consent_category is not same as ('none') %}cookieconsent-optin-{{ cookie_consent_category }}{% endif %}">{{ embed_code|raw }}</div>
+    {% if cookie_consent_category is not same as ('none') %}
+      {# Don't show the cookie disclaimer if "none" is selected. #}
+      <div
+        class="cookieconsent-optout-{{ cookie_consent_category }}">
+        <a href="javascript: Cookiebot.renew()">Acceptér venligst {{ cookie_consent_category|trans }} cookies for at se
+          dette indhold</a>
```
Comment (rasben) [inline web/themes/custom/mungo/templates/paragraphs/paragraph--embed.html.twig:62]: missing translation

**Review by rasben — CHANGES_REQUESTED** [review]
Looks good :) A couple of findings, but nothing major.

P.S. I dont see any translation updates? (da.po / add_translation())

File: configuration/drupal/field.storage.paragraph.field_situation_color.yml [hunk 7]
```diff
@@ -23,6 +23,18 @@ settings:
     -
       value: grey
       label: Grey
+    -
+      value: track-skills-bg
+      label: 'Track - skills (purple)'
```
[the remark this answers is not in the document]
Comment (mikkelblyme) [reply configuration/drupal/field.storage.paragraph.field_situation_color.yml:28]: Good call. Retter til dansk

File: web/sites/default/docker.settings.php [hunk 8]
```diff
@@ -45,10 +45,10 @@
 $config['system.performance']['js']['preprocess'] = FALSE;
 
 // Disable the render cache (this includes the page cache).
-$settings['cache']['bins']['render'] = 'cache.backend.null';
+# $settings['cache']['bins']['render'] = 'cache.backend.null';
```
[the remark this answers is not in the document]
Comment (mikkelblyme) [reply web/sites/default/docker.settings.php:48]: No, on purpose. We mimic prod by default

File: configuration/drupal/core.entity_form_display.node.event.default.yml [hunk 9]
```diff
@@ -69,7 +70,6 @@ third_party_settings:
       weight: 0
       format_type: fieldset
       format_settings:
-        label: 'Event presentation'
```
[the remark this answers is not in the document]
Comment (mikkelblyme) [reply configuration/drupal/core.entity_form_display.node.event.default.yml:72]: https://github.com/reload/ddsdk2016/blob/master/configuration/drupal/core.entity_form_display.node.event.default.yml#L66

**Review by rasben — APPROVED** [review]

**Comment by mikkelblyme** [comment]
backstop-check approve

**Pull request merged**